Failing to build :cinterop:buildJVMSharedLibs

Something in the build setup has changed in a way that makes `:cinterop:buildJVMSharedLibs` fail with the below issue on a M1. 

It is possible to build by first triggering 
```
./gradlew :cinterop:buildJVMSharedLibs
./gradlew :test-base:jvmTest
```
But triggering 
```
./gradlew :test-base:jvmTest
```
alone from a clean slate generates the error, so something is not propagated consistenly throught. 

There are other issues here ... as it seems like we are not clearing stuff properly on `./gradlew clean`. At least 
```
./cinterop/.cxx
./cinterop/src/jvmMain/resources/jni/macos/
``` 
Build error:
```
[ 98%] Building CXX object CMakeFiles/realmc.dir/jni/java_global_ref_by_move.cpp.o
[ 98%] Building CXX object CMakeFiles/realmc.dir/jni/java_class_global_def.cpp.o
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6286:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr4, (jlong **)&arg4, jarg4)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6310:5: error: no matching function for call to 'SWIG_JavaArrayArgoutLonglong'
    SWIG_JavaArrayArgoutLonglong(jenv, jarr4, (jlong *)arg4, jarg4);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1313:13: note: candidate function not viable: no known conversion from 'jlong *' (aka 'long *') to 'long long *' for 3rd argument
static void SWIG_JavaArrayArgoutLonglong (JNIEnv *jenv, jlong *jarr, long long *carr, jlongArray input) {
            ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6425:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr5, (jlong **)&arg5, jarg5)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6448:5: error: no matching function for call to 'SWIG_JavaArrayArgoutLonglong'
    SWIG_JavaArrayArgoutLonglong(jenv, jarr5, (jlong *)arg5, jarg5);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1313:13: note: candidate function not viable: no known conversion from 'jlong *' (aka 'long *') to 'long long *' for 3rd argument
static void SWIG_JavaArrayArgoutLonglong (JNIEnv *jenv, jlong *jarr, long long *carr, jlongArray input) {
            ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6480:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr3, (jlong **)&arg3, jarg3)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6493:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr5, (jlong **)&arg5, jarg5)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6516:5: error: no matching function for call to 'SWIG_JavaArrayArgoutLonglong'
    SWIG_JavaArrayArgoutLonglong(jenv, jarr3, (jlong *)arg3, jarg3);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1313:13: note: candidate function not viable: no known conversion from 'jlong *' (aka 'long *') to 'long long *' for 3rd argument
static void SWIG_JavaArrayArgoutLonglong (JNIEnv *jenv, jlong *jarr, long long *carr, jlongArray input) {
            ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6528:5: error: no matching function for call to 'SWIG_JavaArrayArgoutLonglong'
    SWIG_JavaArrayArgoutLonglong(jenv, jarr5, (jlong *)arg5, jarg5);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1313:13: note: candidate function not viable: no known conversion from 'jlong *' (aka 'long *') to 'long long *' for 3rd argument
static void SWIG_JavaArrayArgoutLonglong (JNIEnv *jenv, jlong *jarr, long long *carr, jlongArray input) {
            ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6724:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr3, (jlong **)&arg3, jarg3)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6747:5: error: no matching function for call to 'SWIG_JavaArrayArgoutLonglong'
    SWIG_JavaArrayArgoutLonglong(jenv, jarr3, (jlong *)arg3, jarg3);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1313:13: note: candidate function not viable: no known conversion from 'jlong *' (aka 'long *') to 'long long *' for 3rd argument
static void SWIG_JavaArrayArgoutLonglong (JNIEnv *jenv, jlong *jarr, long long *carr, jlongArray input) {
            ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6845:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr2, (jlong **)&arg2, jarg2)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:6869:5: error: no matching function for call to 'SWIG_JavaArrayArgoutLonglong'
    SWIG_JavaArrayArgoutLonglong(jenv, jarr2, (jlong *)arg2, jarg2);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1313:13: note: candidate function not viable: no known conversion from 'jlong *' (aka 'long *') to 'long long *' for 3rd argument
static void SWIG_JavaArrayArgoutLonglong (JNIEnv *jenv, jlong *jarr, long long *carr, jlongArray input) {
            ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:7113:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr3, (jlong **)&arg3, jarg3)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:7136:5: error: no matching function for call to 'SWIG_JavaArrayArgoutLonglong'
    SWIG_JavaArrayArgoutLonglong(jenv, jarr3, (jlong *)arg3, jarg3);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1313:13: note: candidate function not viable: no known conversion from 'jlong *' (aka 'long *') to 'long long *' for 3rd argument
static void SWIG_JavaArrayArgoutLonglong (JNIEnv *jenv, jlong *jarr, long long *carr, jlongArray input) {
            ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:7517:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr3, (jlong **)&arg3, jarg3)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:7540:5: error: no matching function for call to 'SWIG_JavaArrayArgoutLonglong'
    SWIG_JavaArrayArgoutLonglong(jenv, jarr3, (jlong *)arg3, jarg3);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1313:13: note: candidate function not viable: no known conversion from 'jlong *' (aka 'long *') to 'long long *' for 3rd argument
static void SWIG_JavaArrayArgoutLonglong (JNIEnv *jenv, jlong *jarr, long long *carr, jlongArray input) {
            ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:7590:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr2, (jlong **)&arg2, jarg2)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:7613:5: error: no matching function for call to 'SWIG_JavaArrayArgoutLonglong'
    SWIG_JavaArrayArgoutLonglong(jenv, jarr2, (jlong *)arg2, jarg2);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1313:13: note: candidate function not viable: no known conversion from 'jlong *' (aka 'long *') to 'long long *' for 3rd argument
static void SWIG_JavaArrayArgoutLonglong (JNIEnv *jenv, jlong *jarr, long long *carr, jlongArray input) {
            ^
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:7699:10: error: no matching function for call to 'SWIG_JavaArrayInLonglong'
    if (!SWIG_JavaArrayInLonglong(jenv, &jarr3, (jlong **)&arg3, jarg3)) return 0;
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp:1292:12: note: candidate function not viable: no known conversion from 'jlong **' (aka 'long **') to 'long long **' for 3rd argument
static int SWIG_JavaArrayInLonglong (JNIEnv *jenv, jlong **jarr, long long **carr, jlongArray input) {
           ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [CMakeFiles/realmc.dir/Users/claus.rorbech/proj/realm-kotlin-work/packages/jni-swig-stub/build/generated/sources/jni/realmc.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/realmc.dir/all] Error 2
make: *** [all] Error 2

> Task :cinterop:buildJVMSharedLibs FAILED
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,7 +22,7 @@
 * Minimum Android SDK: 16.
 
 ### Internal
-* None.
+* Updated to require Swig 4.1.0.
 
 
 ## 1.5.1 (YYYY-MM-DD)

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -11,7 +11,7 @@ We welcomes all contributions! The only requirement we have is that, like many o
 
 ### Prerequisites
 
-- Swig. On Mac this can be installed using Homebrew: `brew install swig`.
+- Swig 4.1.0 or above. On Mac this can be installed using Homebrew: `brew install swig`.
 - Ccache. On Mac this can be installed using Homebrew: `brew install ccache`.
 - CMake 3.18.1 or above. Can be installed through the Android SDK Manager.
 - Java 11.

--- a/packages/jni-swig-stub/realm.i
+++ b/packages/jni-swig-stub/realm.i
@@ -361,30 +361,39 @@ bool throw_as_java_exception(JNIEnv *jenv) {
 // Enable passing output argument pointers as long[]
 %apply int64_t[] {void **};
 // Type map for int64_t has an erroneous cast, don't know how to fix it except with this
-//%typemap(in) void** ( jlong *jarr ){
-//    // Original
-//    %#if defined(__ANDROID__) && defined(__aarch64__) // Android arm64-v8a
-//        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long **)&$1, $input)) return $null;
-//    %#elif defined(__ANDROID__) // Android armeabi-v7a, x86_64 and x86
-//        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
-//    %#elif defined(__aarch64__) // macos M1
-//        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
-//    %#else
-//        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long long **)&$1, $input)) return $null;
-//    %#endif
-//}
-//%typemap(argout) void** {
-//    // Original
-//    %#if defined(__ANDROID__) && defined(__aarch64__)
-//        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long*)$1, $input);
-//    %#elif defined(__ANDROID__)
-//        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong*)$1, $input);
-//    %#elif defined(__aarch64__)
-//        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong *)$1, $input);
-//    %#else
-//        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long long *)$1, $input);
-//    %#endif
-//}
+%typemap(in) void** ( jlong *jarr ){
+#if SWIG_VERSION>=0x040100
+    // Original
+    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long long **)&$1, $input)) return $null;
+#else
+    %#if defined(__ANDROID__) && defined(__aarch64__) // Android arm64-v8a
+    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long **)&$1, $input)) return $null;
+    %#elif defined(__ANDROID__) // Android armeabi-v7a, x86_64 and x86
+    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
+    %#elif defined(__aarch64__) // macos M1
+    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
+    %#else
+    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long long **)&$1, $input)) return $null;
+    %#endif
+#endif
+}
+%typemap(argout) void** {
+#if SWIG_VERSION>=0x040100
+    // Original
+    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long long *)$1, $input);
+#else
+    %#if defined(__ANDROID__) && defined(__aarch64__)
+    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long*)$1, $input);
+    %#elif defined(__ANDROID__)
+    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong*)$1, $input);
+    %#elif defined(__aarch64__)
+    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong *)$1, $input);
+    %#else
+    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long long *)$1, $input);
+    %#endif
+#endif
+}
+
 %apply void** {realm_object_t**, realm_list_t**, size_t*, realm_class_key_t*,
                realm_property_key_t*, realm_user_t**, realm_set_t**};
 

--- a/packages/jni-swig-stub/realm.i
+++ b/packages/jni-swig-stub/realm.i
@@ -360,40 +360,6 @@ bool throw_as_java_exception(JNIEnv *jenv) {
 
 // Enable passing output argument pointers as long[]
 %apply int64_t[] {void **};
-// Type map for int64_t has an erroneous cast, don't know how to fix it except with this
-%typemap(in) void** ( jlong *jarr ){
-#if SWIG_VERSION>=0x040100
-    // Original
-    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long long **)&$1, $input)) return $null;
-#else
-    %#if defined(__ANDROID__) && defined(__aarch64__) // Android arm64-v8a
-    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long **)&$1, $input)) return $null;
-    %#elif defined(__ANDROID__) // Android armeabi-v7a, x86_64 and x86
-    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
-    %#elif defined(__aarch64__) // macos M1
-    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
-    %#else
-    if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long long **)&$1, $input)) return $null;
-    %#endif
-#endif
-}
-%typemap(argout) void** {
-#if SWIG_VERSION>=0x040100
-    // Original
-    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long long *)$1, $input);
-#else
-    %#if defined(__ANDROID__) && defined(__aarch64__)
-    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long*)$1, $input);
-    %#elif defined(__ANDROID__)
-    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong*)$1, $input);
-    %#elif defined(__aarch64__)
-    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong *)$1, $input);
-    %#else
-    SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long long *)$1, $input);
-    %#endif
-#endif
-}
-
 %apply void** {realm_object_t**, realm_list_t**, size_t*, realm_class_key_t*,
                realm_property_key_t*, realm_user_t**, realm_set_t**};
 

--- a/packages/jni-swig-stub/realm.i
+++ b/packages/jni-swig-stub/realm.i
@@ -361,30 +361,30 @@ bool throw_as_java_exception(JNIEnv *jenv) {
 // Enable passing output argument pointers as long[]
 %apply int64_t[] {void **};
 // Type map for int64_t has an erroneous cast, don't know how to fix it except with this
-%typemap(in) void** ( jlong *jarr ){
-    // Original
-    %#if defined(__ANDROID__) && defined(__aarch64__) // Android arm64-v8a
-        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long **)&$1, $input)) return $null;
-    %#elif defined(__ANDROID__) // Android armeabi-v7a, x86_64 and x86
-        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
-    %#elif defined(__aarch64__) // macos M1
-        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
-    %#else
-        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long long **)&$1, $input)) return $null;
-    %#endif
-}
-%typemap(argout) void** {
-    // Original
-    %#if defined(__ANDROID__) && defined(__aarch64__)
-        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long*)$1, $input);
-    %#elif defined(__ANDROID__)
-        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong*)$1, $input);
-    %#elif defined(__aarch64__)
-        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong *)$1, $input);
-    %#else
-        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long long *)$1, $input);
-    %#endif
-}
+//%typemap(in) void** ( jlong *jarr ){
+//    // Original
+//    %#if defined(__ANDROID__) && defined(__aarch64__) // Android arm64-v8a
+//        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long **)&$1, $input)) return $null;
+//    %#elif defined(__ANDROID__) // Android armeabi-v7a, x86_64 and x86
+//        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
+//    %#elif defined(__aarch64__) // macos M1
+//        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (jlong **)&$1, $input)) return $null;
+//    %#else
+//        if (!SWIG_JavaArrayInLonglong(jenv, &jarr, (long long **)&$1, $input)) return $null;
+//    %#endif
+//}
+//%typemap(argout) void** {
+//    // Original
+//    %#if defined(__ANDROID__) && defined(__aarch64__)
+//        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long*)$1, $input);
+//    %#elif defined(__ANDROID__)
+//        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong*)$1, $input);
+//    %#elif defined(__aarch64__)
+//        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (jlong *)$1, $input);
+//    %#else
+//        SWIG_JavaArrayArgoutLonglong(jenv, jarr$argnum, (long long *)$1, $input);
+//    %#endif
+//}
 %apply void** {realm_object_t**, realm_list_t**, size_t*, realm_class_key_t*,
                realm_property_key_t*, realm_user_t**, realm_set_t**};
 

